File: Interface Gráfica - Tela de Criação de Herói/Java/Guerreiro.java
class Guerreiro extends Personagem {
    protected int defesa = 5;

    public Guerreiro(String nome, int vida, int nivel, int defesa) {
        super(nome, vida, nivel);
        setdefesa(defesa);
    }

    // Getters
    public int getdefesa() {
        return (defesa);
    }

    // Setters
    public void setdefesa(int defesa) {
        if (defesa >= 0) {
            this.defesa = defesa;
        } else {
            throw new IllegalArgumentException(
                "Defesa inválida! Insira um valor maior ou igual a 0 (valor informado: " + defesa + ")"
            );
        }
    }

    @Override
    public int receberDano(int dano) {
        return super.receberDano(Math.max(0, dano - defesa));
    }

    @Override
    public void ficha() {
        super.ficha();
        System.out.printf("%nDefesa: %d%n", defesa);
    }

    @Override
    public String habilidade() {
        return nome + " usa Escudo de Aço";
    }
}

File: Exceções - IF Quest/Python/guerreiro.py
from personagem import Personagem


class Guerreiro(Personagem):
    def __init__(self, nome, vida, nivel, defesa):
        super().__init__(nome, vida, nivel)
        self.defesa = 5
        self.setdefesa(defesa)

    # Getters
    def getdefesa(self):
        return self.defesa

    # Setters
    def setdefesa(self, defesa):
        if defesa >= 0:
            self.defesa = defesa
        else:
            raise ValueError(
                f"Defesa inválida! Insira um valor maior ou igual a 0 (valor informado: {defesa})"
            )

    def receberDano(self, dano):
        return super().receberDano(max(0, dano - self.defesa))

    def ficha(self):
        super().ficha()
        print(f"\nDefesa: {self.defesa}")

    def habilidade(self):
        return f"{self.nome} usa Escudo de Aço"
